ahbottomnavigation/src/main/java/com/aurelhubert/ahbottomnavigation/notification/AHNotification.java

```diff
@@ -0,0 +1,117 @@
+package com.aurelhubert.ahbottomnavigation.notification;
+
+import android.os.Parcel;
+import android.os.Parcelable;
+import android.support.annotation.ColorInt;
+import android.support.annotation.Nullable;
+import android.text.TextUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author repitch
+ */
+public class AHNotification implements Parcelable {
+
+    @Nullable
+    private String text; // can be null, so notification will not be shown
+
+    @ColorInt
+    private int textColor; // if 0 then use default value
+
+    @ColorInt
+    private int backgroundColor; // if 0 then use default value
+
+    private AHNotification() {
+        // empty
+    }
+
+    private AHNotification(Parcel in) {
+        text = in.readString();
+        textColor = in.readInt();
+        backgroundColor = in.readInt();
+    }
+
+    public boolean isEmpty() {
+        return TextUtils.isEmpty(text);
+    }
+
+    public String getText() {
+        return text;
+    }
+
+    public int getTextColor() {
+        return textColor;
+    }
+
+    public int getBackgroundColor() {
+        return backgroundColor;
+    }
+
+    public static AHNotification justText(String text) {
+        return newBuilder().setText(text).build();
+    }
+
+    public static Builder newBuilder() {
+        return new AHNotification().new Builder();
+    }
+
+    public static List<AHNotification> generateEmptyList(int size) {
+        List<AHNotification> notificationList = new ArrayList<>();
+        for (int i = 0; i < size; i++) {
+            notificationList.add(AHNotification.newBuilder().build());
+        }
+        return notificationList;
+    }
+
+    @Override
+    public int describeContents() {
+        return 0;
+    }
+
+    @Override
+    public void writeToParcel(Parcel dest, int flags) {
+        dest.writeString(text);
+        dest.writeInt(textColor);
+        dest.writeInt(backgroundColor);
+    }
+
+    public class Builder {
+        private Builder() {
+            // empty
+        }
+
+        public Builder setText(String text) {
+            AHNotification.this.text = text;
+            return this;
+        }
+
+        public Builder setTextColor(@ColorInt int textColor) {
+            AHNotification.this.textColor = textColor;
+            return this;
+        }
+
+        public Builder setBackgroundColor(@ColorInt int backgroundColor) {
+            AHNotification.this.backgroundColor = backgroundColor;
+            return this;
+        }
+
+        public AHNotification build() {
```

Comment:
Builder alsway has reference to AHNotification instance. After call build() method you can change AHNotification state using builder methods. Use builder as static class and create new AHNotification instance in build() method.